src/flavors/MC/MC.jl

```diff
@@ -234,5 +317,203 @@ Performs a sweep of local moves.
+
+    mc.a = MCAnalysis()
+
+    _time = time()
+    verbose && println("\n\nReplaying measurement stage - ", length(configs))
+    prepare!(mc.measurements, mc, mc.model)
+    for i in start:mc.p.measure_rate:length(configs)
+        mc.config = configs[i]
+        # maybe calculate total Energy?
+        measure!(mc.measurements, mc, mc.model, i)
+
+        if mc.p.print_rate != 0 && iszero(mod(i, mc.p.print_rate))
+            sweep_dur = (time() - _time)/print_rate
+            max_sweep_duration = max(max_sweep_duration, sweep_dur)
+            if verbose
+                println("\t", i)
+                @printf("\t\tsweep dur: %.3fs\n", sweep_dur)
+            end
+            flush(stdout)
+            _time = time()
+        end
+
+        if safe_before - now() < Millisecond(grace_period) +
+                Millisecond(round(Int, 2e3max_sweep_duration))
+
+            println("Early save initiated for sweep #$i.\n")
+            verbose && println("Current time: ", Dates.format(now(), "d.u yyyy HH:MM"))
+            verbose && println("Target time:  ", Dates.format(safe_before, "d.u yyyy HH:MM"))
+            filename = save(filename, mc, force_overwrite = force_overwrite)
+            save_rng(filename)
+            jldopen(filename, "r+") do f
+                write(f, "last_sweep", i)
+            end
+            verbose && println("\nEarly save finished")
+
+            return false
+        end
+    end
+    finish!(mc.measurements, mc, mc.model)
+
+    end_time = now()
+    verbose && println("Ended: ", Dates.format(end_time, "d.u yyyy HH:MM"))
+    verbose && @printf("Duration: %.2f minutes", (end_time - start_time).value/1000. /60.)
+
+    return true
+end
+
+
+#     save_mc(filename, mc, entryname)
+#
+# Saves (minimal) information necessary to reconstruct a given `mc::MC` to a
+# JLD-file `filename` under group `entryname`.
+#
+# When saving a simulation the default `entryname` is `MC`
+function save_mc(file::JLD.JldFile, mc::MC, entryname::String="MC")
+    write(file, entryname * "/VERSION", 0)
+    write(file, entryname * "/type", typeof(mc))
+    save_parameters(file, mc.p, entryname * "/parameters")
+    write(file, entryname * "/conf", mc.conf)
+    save_measurements(file, mc, entryname * "/Measurements")
+    save_model(file, mc.model, entryname * "/Model")
+    nothing
+end
+
+#     load_mc(data, ::Type{<: MC})
+#
+# Loads a MC from a given `data` dictionary produced by `JLD.load(filename)`.
+function load_mc(data, ::Type{T}) where {T <: MC}
+    @assert data["VERSION"] == 0
+    mc = data["type"]()
+    mc.p = load_parameters(data["parameters"], data["parameters"]["type"])
+    mc.conf = data["conf"]
+    mc.model = load_model(data["Model"], data["Model"]["type"])
+
+    measurements = load_measurements(data["Measurements"])
+    mc.thermalization_measurements = measurements[:TH]
+    mc.measurements = measurements[:ME]
+    mc
+end
+
+
+#   save_parameters(file::JLD.JldFile, p::MCParameters, entryname="Parameters")
```

Comment:
Why are these comments instead of docstrings? (Here and for many other functions)

Reply:
In a lot of cases I started writing doc strings, then thought "this is not something that should be used from the outside" and made them comments